keps/sig-api-machinery/3157-watch-list/README.md

```diff
@@ -852,21 +868,157 @@ will rollout across nodes.
 
 ###### What specific metrics should inform a rollback?
 
+`apiserver_terminated_watchers_total` - a large number of terminated watchers might indicate synchronization issues.
+For example, we have some client-side error where we're not getting data from the server. Or we have a server-side error, and the buffer is getting cluttered.
+
+`apiserver_request_duration_second_bucket` - in general, a large number of "short" watch requests can indicate synchronization issues.
+
+`apiserver_watch_list_duration_seconds` - the absence of this metric may indicate that the client did not receive a special bookmark.
+The issue here could be that the server never sent it due to an error or didn't even receive it from the database.
+
+`apiserver_watch_list_duration_seconds` - long synchronization times may indicate that the server is lagging behind etcd. 
+Forr example, not receiving progress notifications from the database frequently.
+
+`apiserver_watch_cache_lag` - tells how far behind the server is compared to the database.
+Significant discrepancies affect the times for full data synchronization.
+
+A good metric can also be the number of kube-controller-manager restarts. 
+Which may indicate issues with informers synchronization.
+
 <!--
 What signals should users be paying attention to when the feature is young
 that might indicate a serious problem?
 -->
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+Upgrade->downgrade->upgrade testing was done manually using the following steps:
```

Comment:
This path assumes https://github.com/kubernetes/kubernetes/pull/120939 and https://github.com/kubernetes/kubernetes/pull/120923 have been merged.